odl/discr/discr_mappings.py

```diff
@@ -233,71 +235,67 @@ def _call(self, inp):
         >>> from odl import FunctionSet, RealNumbers
         >>> funcset = FunctionSet(grid.convex_hull(), RealNumbers())
 
-        Finally create the operator:
+        Finally create the operator and test it on a function:
 
-        >>> coll_op = RawGridCollocation(funcset, grid, rn)
-        >>> func_elem = funcset.element(lambda x, y: x - y)
+        >>> coll_op = GridCollocation(funcset, grid, rn)
+        ...
+        >>> def func(x):
+        ...     return x[0] - x[1]  # properly vectorized
+        ...
+        >>> func_elem = funcset.element(func)
         >>> coll_op(func_elem)
         Rn(6).element([-2.0, -3.0, -4.0, -1.0, -2.0, -3.0])
+        >>> coll_op(func)  # Works directly
+        Rn(6).element([-2.0, -3.0, -4.0, -1.0, -2.0, -3.0])
+        >>> out = Rn(6).element()
+        >>> coll_op(func, out=out)  # In-place
+        Rn(6).element([-2.0, -3.0, -4.0, -1.0, -2.0, -3.0])
 
-        Or, if we want Fortran ordering:
+        Fortran ordering:
 
-        >>> coll_op = RawGridCollocation(funcset, grid, rn, order='F')
+        >>> coll_op = GridCollocation(funcset, grid, rn, order='F')
         >>> coll_op(func_elem)
         Rn(6).element([-2.0, -1.0, -3.0, -2.0, -4.0, -3.0])
         """
-        try:
-            mg_tuple = self.grid.meshgrid()
-            values = inp(*mg_tuple).ravel(order=self.order)
-        except TypeError:
-            points = self.grid.points(order=self.order)
-            values = np.empty(points.shape[0], dtype=self.range.dtype)
-            for i, point in enumerate(points):
-                values[i] = inp(*point)
-        return self.range.element(values)
+        mg = self.grid.meshgrid()
+        if out is None:
+            out = func(mg).ravel(order=self.order)
+        else:
+            func(mg, out=out.asarray().reshape(self.grid.shape,
+                                               order=self.order))
+        return out
 
 
-class GridCollocation(RawGridCollocation):
+class NearestInterpolation(FunctionSetMapping):
 
-    """Function evaluation at grid points.
+    """Nearest neighbor interpolation as an `Operator`.
 
-    This is the linear `Operator` version of the default
-    'restriction' used by all core discretization classes.
-    """
+    Given points :math:`x_1, \dots, x_n \\in \mathbb{R}` and values
+    :math:`f_1, \dots, f_n \\in \mathbb{F}`, the nearest neighbor
+    interpolation at an arbitrary point :math:`x \\in \mathbb{R}`
+    is defined by
 
-    def __init__(self, ip_fspace, grid, dspace, order='C'):
-        """Initialize a new instance.
+        :math:`I_{\\bar f}(x) := x_j, \\text{ where } j
```

Comment:
For me it works without problems. And in that case, I'd prefer this version since it also works for the "Object inspector" of Spyder. Of course, cutting down on math formulas in docstrings is still a valid goal.